tools/plot-bench.py

```diff
@@ -0,0 +1,305 @@
+    """
+    if args is None:
+        return None
+    
+    return set(map(lambda arg: arg.strip(), args.split(',')))
+
+
+def make_plot_jobs_and_directories(
+    programs: Set[str],
+    plot_jobs: Dict[str, Dict[str, Dict[str, Any]]],
+    data: Dict[str, Dict[str, Dict[str, Any]]],
+    plots: Set[str],
+    filetype: str
+    ) -> None:
+    """Makes dictionary with plot jobs where plot_jobs are the jobs."""
+
+    for program in programs:
+        datasets = data.get(program).get('datasets')
+        for dataset_path, dataset in datasets.items():
+            runtimes = dataset.get('runtimes')
+            directory = os.path.dirname(program)
+            program_basename = pathlib.Path(program).stem
+            dataset_basename = pathlib.Path(dataset_path).stem
+            dataset_name = textwrap.shorten(dataset_basename, 50).replace(' ', '-')
+            os.makedirs(directory, exist_ok=True)
+            for plot in plots:
+                filename = f'{program_basename}-{dataset_name}-{plot}.{filetype}'
+                plot_file = os.path.join(directory, filename)
+                plot_jobs[plot][plot_file] = {
+                    'program': program_basename,
+                    'dataset': dataset_basename,
+                    'runtimes': np.array(runtimes)
+                }
+
+def set_quality(quality: str) -> int:
+    """The quality is set based on the string given and the dpi is returned."""
+
+    if quality == 'low':
+        return 100
+    elif quality == 'medium':
+        return 200
+    elif quality == 'high':
+        plt.rcParams.update({
+            'text.usetex': True,
+            'font.family': 'serif',
+            'font.serif': ['Computer Modern Serif']
+        })
+        return 400
+    
+    raise Exception(f'{quality} is an unknown quality.')
+
+def main():
+    plt.rcParams.update({
+        'ytick.color': 'black',
+        'xtick.color': 'black',
+        'axes.labelcolor': 'black',
+        'axes.edgecolor': 'black',
+        'axes.axisbelow': True,
+        'text.usetex': False,
+        'axes.prop_cycle': matplotlib.cycler(color=['#5f021f'])
+    })
+
+    args = get_args()
+    data = json.load(open(args.filename, 'r'))
+    plots = format_arg_list(args.plots)
+    programs = format_arg_list(args.programs)
+    filetype = args.filetype
+    quality = args.quality
+    transparent = args.transparent
+    matplotlib.use(args.backend)
+    
+    dpi = set_quality(quality)
+    plotters = plotters_factory(dpi=dpi, transparent=transparent)
+    plot_jobs = {k: dict() for k in plotters.keys()}
+
+    if plots is None:
+        plots = set(plot_jobs.keys())
+
+    if programs is None:
+        programs = set(data.keys())
+
+    make_plot_jobs_and_directories(programs, plot_jobs, data, plots, filetype)
+
+    for plot, data in plot_jobs.items():
+        for data_chunk in chunks(data, max(len(data) // 4, 1)):
+            plotters[plot]().plot(data_chunk)
```

Comment:
I would like the program to print the name (to stdout) of every plot file that it creates.